### PR TITLE
Snapshot status api: make sure headers are handed over to inner nodes request

### PR DESCRIPTION
**Pull request by javanna**

### CODE REVIEW

File: src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java [hunk 1]
```diff
@@ -43,6 +43,11 @@ protected NodesOperationRequest() {
 
     }
 
+    protected NodesOperationRequest(ActionRequest request, String... nodesIds) {
+        super(request);
+        this.nodesIds = nodesIds;
+    }
+
     protected NodesOperationRequest(String... nodesIds) {
```
Comment (martijnvg) [inline src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java]: maybe this constructor should be removed?

**Comment by martijnvg** [comment]
left one minor comment, other than that LGTM

File: src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java (hunk 1 above)
Comment (javanna) [inline src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java]: I wish we could remove it, but it's used by other requests that are created directly through java api, for instance nodes hot threads, nodes info etc. those don't need headers in their constructor, as the client automatically copies over the headers in the Rest action.

File: src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java [hunk 2]
```diff
@@ -43,6 +43,11 @@ protected NodesOperationRequest() {
 
     }
 
+    protected NodesOperationRequest(ActionRequest request, String... nodesIds) {
```
Comment (uboness) [inline src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java]: does that mean that all node operation requests suffer from the same issue? did you check for other potential bugs around this?

Comment (javanna) [inline src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java]: all of the other requests that extend this one are user facing requests, this is the only case where we extended for an internal request that actually needed the headers copied from the parent request. We are good ;)

**Pull request closed without merging**